app/src/main/java/com/amaze/filemanager/filesystem/ftp/NetCopyConnectionInfo.kt

```diff
@@ -104,7 +109,18 @@ class NetCopyConnectionInfo(url: String) {
                      * Invalid string would have been trapped to other branches. Strings fell into
                      * this branch must be integer
                      */
-                        it[7].toInt()
+                        try {
+                            // Need to make sure port number is in range
+                            if (it[7].toInt() in 1..65535) {
+                                it[7].toInt()
+                            } else {
+                                LOGGER.warn("Port number is out of range: ${it[7]}")
+                                0
+                            }
+                        } catch (e: NumberFormatException) {
+                            LOGGER.warn("Unable to parse port number: ${it[7]}", e)
+                            0
```

Comment:
See above.

Reply:
Same here. It should be downstream code's responsibility of determining the default port number of different protocols.